File: 01_Functions_and_Execution_Context/04-InvokingFxns.js
// Examine the code given to you. Try to make the calls variable equal to
// 'JerryKramerGeorgeElaine' with only a single invocation to the function jerry.

let calls = "";

function jerry(str) {
  str += "Jerry";
  return george(str);
 }

 function george(str) {
   str += "Kramer";
   return kramer(str);
 }

 function elaine(str) {
   str += "Elaine";
   return str;
 }

 function kramer(str) {
   str += "George";
   return elaine(str);
 }

 // should return: 'JerryKramerGeorgeElaine'
calls = jerry(calls);

console.log(calls);

File: 01_Functions_and_Execution_Context/04-InvokingFxns.py
# Examine the code given to you. Try to make the calls variable equal to
# 'JerryKramerGeorgeElaine' with only a single invocation to the function jerry.

calls = ""


def jerry(str):
    str += "Jerry"
    return george(str)


def george(str):
   str += "Kramer"
   return kramer(str)


def elaine(str):
   str += "Elaine"
   return str


def kramer(str):
   str += "George"
   return elaine(str)

# should return: 'JerryKramerGeorgeElaine'

calls = jerry(calls)

print(calls)